Services/Certificate/classes/LearningHistory/class.ilCertificateLearningHistoryProvider.php

```diff
@@ -1,63 +1,35 @@
-<?php
+<?php declare(strict_types=1);
+
 /* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
 
 use ILIAS\UI\Factory;
 use ILIAS\UI\Renderer;
+use ILIAS\DI\Container;
 
 /**
  * @author  Niels Theen <ntheen@databay.de>
  */
 class ilCertificateLearningHistoryProvider extends ilAbstractLearningHistoryProvider implements ilLearningHistoryProviderInterface
 {
-    /**
-     * @var ilUserCertificateRepository
-     */
-    private $userCertificateRepository;
+    private ?ilUserCertificateRepository $userCertificateRepository;
+    private ?ilCtrl $controller;
+    private ?ilSetting $certificateSettings;
+    protected ?Factory $uiFactory;
+    protected ?Renderer $uiRenderer;
+    private ?ilCertificateUtilHelper $utilHelper;
```

Comment:
This is not optional. Only the passed constructor argument is optional, the value stored in `$this->utilHelper` MUST never be `null`.